src/backend/distributed/commands/function.c

```diff
@@ -631,6 +632,57 @@ TriggerSyncMetadataToPrimaryNodes(void)
 }
 
 
+/*
+ * ShouldPropagateCreateFunction tests if we need to propagate a CREATE FUNCTION
+ * statement. We only propagate replace's of distributed functions to keep the function on
+ * the workers in sync with the one on the coordinator.
+ */
+static bool
+ShouldPropagateCreateFunction(CreateFunctionStmt *stmt)
+{
+	const ObjectAddress *address = NULL;
+
+	if (creating_extension)
+	{
+		/*
+		 * extensions should be created separately on the workers, functions cascading
+		 * from an extension should therefore not be propagated.
+		 */
+		return false;
+	}
+
+	if (!EnableDependencyCreation)
+	{
+		/*
+		 * we are configured to disable object propagation, should not propagate anything
+		 */
+		return false;
+	}
+
+	if (!stmt->replace)
```

Comment:
I guess we shouldn't need this check because `if (!stmt->replace)`, it cannot be a distributed function -- so the below check would return false.

On the other hand, this makes it easier to follow the code -- more explicit code. So, keeping it seems like a good option as well

Reply:
One could do a create without a replace on a distributed function, this would error during local execution though. I think having this explicitly here is more clear towards the intention of the function.